src/components/forms/PlanForm/index.tsx

```diff
@@ -173,6 +182,8 @@ export type LocationChildRenderProp = (
 
 /** interface for plan form props */
 export interface PlanFormProps {
+  actionCreator: ActionCreator<FetchPlanRecordsAction>;
+  plansArray: PlanRecord[];
```

Comment:
How are you using this plans array?  I don't see why PlanForm needs to receive an array of plans?

Reply:
@moshthepitt, On submitting new plan, We need this array so we can push new plan in it and then update the store.

Reply:
Okay cool.  I think we don't need the entire array.  The store would work fine if we added just one plan definition.  Either by using an array of one element, or by adding an action creator that accepts a single plan.